src/main.rs

```diff
@@ -0,0 +1,81 @@
+use chrono::{Local, NaiveTime, TimeDelta};
+use serde::Deserialize;
+use tokio::{task::JoinSet, time::sleep};
+use url::Url;
+
+extern crate tokio;
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct Device {
+    name: String,
+    on: Url,
+    off: Url,
+    events: Vec<Event>,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct Event {
+    timestamp: NaiveTime,
+    on: bool,
+}
+
+#[tokio::main]
+async fn main() {
+    let devices: Vec<Device> = serde_json::from_str(
+        &(tokio::fs::read_to_string("the.json")
+            .await
+            .expect("reading in JSON to work")),
+    )
+    .expect("JSON resolves to Vec<Device>");
+    println!("devices as Vec<Device> : {:?}", devices);
+
+    let mut set: JoinSet<()> = JoinSet::new();
+
+    for device in devices {
+        println!("working with device {:?}", device.name);
+        let evs = device.events.clone();
```

Comment:
Nit: probably don't need to clone here, just iterate over the original events vec.